server/service/src/document/patient/insert.rs

```diff
@@ -0,0 +1,82 @@
+use chrono::Utc;
+use repository::{EqualFilter, RepositoryError};
+use util::uuid::uuid;
+
+use crate::{
+    document::{
+        document_service::{DocumentInsertError, DocumentService, DocumentServiceTrait},
+        raw_document::RawDocument,
+    },
+    service_provider::ServiceContext,
+};
+
+use super::{
+    patient_doc_name, patient_schema::SchemaPatient, Patient, PatientFilter, PatientService,
+    PatientServiceTrait, PATIENT_TYPE,
+};
+
+#[derive(PartialEq, Debug)]
+pub enum InsertPatientError {
+    InvalidDataSchema(Vec<String>),
+    InternalError(String),
+    DatabaseError(RepositoryError),
+}
+
+pub struct InsertPatient {
+    pub data: serde_json::Value,
+    pub schema_id: Option<String>,
+}
+
+pub fn insert_patients(
+    ctx: &ServiceContext,
+    store_id: String,
+    user_id: &str,
+    input: InsertPatient,
+) -> Result<Patient, InsertPatientError> {
+    let new_patient_id = uuid();
+
+    // update patient id
+    let mut patient: SchemaPatient = serde_json::from_value(input.data).map_err(|err| {
+        InsertPatientError::InvalidDataSchema(vec![format!("Invalid patient data: {}", err)])
+    })?;
+    patient.id = new_patient_id.clone();
```

Comment:
actually, looking at it agiain, we deserialise, then change id, then serialise, just to pass patient_id to update_document...

Reply:
This is to check that the schema is valid (even thought could be enforced earlier) and to have type safety when setting the id. Think its nice to have the id in the Patient document (even if a document would have an associated patient)

Reply:
I think it's very hacky to ` deserialise, then change id, then serialise, just to pass patient_id to update_document`

> Think its nice to have the id in the Patient document (even if a document would have an associated patient)

Why ?

Reply:
Otherwise the patient document wouldn't be complete because it doesn't contain its own id. But I see what you mean. Any suggestions? Do you mean to not include the mSupply patient id in the document but manage it outside of it?